### PR TITLE
develop serverside nodes

### PR DESCRIPTION
**Pull request by hidetak**
This is an implementation of the server-side node described in the following Issue.
https://github.com/thingweb/node-red-contrib-web-of-things/issues/25

The following modifications are also implemented for the client side.
* modified to use node-wot 0.8.10
* read property node execution is now triggered by input
* Wait until the observe property and subscribe event execution succeeds

### CODE REVIEW

**Comment by egekorkan** [comment]
We discussed this in the call and @egekorkan will try using them on his computer. Then we can merge it and then move to the Eclipse Thingweb.

The issues in node-wot are:
- https://github.com/eclipse-thingweb/node-wot/issues/1241
- https://github.com/eclipse-thingweb/node-wot/issues/1237
- https://github.com/eclipse-thingweb/node-wot/issues/1233

We can merge this PR without fixing those first.

**Comment by danielpeintner** [comment]
> * modified to use node-wot 0.8.10

Minor: there is already node-wot v0.8.12 out. Updating to it shouldn't cause any issue since there are no API changes or such. Thanks!

File: package.json [hunk 1]
```diff
@@ -14,24 +17,40 @@
   },
   "repository": {
     "type": "git",
-    "url": "https://github.com/thingweb/node-red-contrib-web-of-things"
+    "url": "https://github.com/hidetak/node-red-contrib-web-of-things"
```
Comment (egekorkan) [inline package.json]: Could you keep these as is?

File: package.json [hunk 2]
```diff
@@ -14,24 +17,40 @@
...
   },
-  "homepage": "https://github.com/thingweb/node-red-contrib-web-of-things/blob/master/README.md",
+  "homepage": "https://github.com/hidetak/node-red-contrib-web-of-things/blob/master/README.md",
```
Comment (egekorkan) [inline package.json]: Could you keep these as is?

**Review by egekorkan — CHANGES_REQUESTED** [review]
Small changes but overall, it is working. I just didn't manage to get the property on the server side working. I have just put the following in the function attached to the property:

image

@hidetak how can I make it work? I get an empty payload when I read the property.

Overall, thank you many times for doing this work! You even have Japanese UI elements that will make Thingweb project international ;)

**Comment by hidetak** [comment]
@danielpeintner 

Thank you for your comment.
I have changed the version of node-wot to 0.8.12.

File: package.json (hunk 1 above)
Comment (hidetak) [reply package.json]: Sorry, I forgot to correct the URL.
I have corrected it.

File: package.json (hunk 2 above)
Comment (hidetak) [reply package.json]: Sorry, I forgot to correct the URL.
I have corrected it.

**Comment by hidetak** [comment]
@egekorkan 

Thank you for your confirmation.

image

The input of a Server-Property node is done when there is a change to that property on the server side. At this time, it is not necessary to input the changed value.

The value returned to the client must be obtained by a flow connected to the first output of the Server-Property node and input to the Server-End node.

The value of the write is output from the second output of the Server-Property node, so the connected flow should process that value and the end should connect the Server-End node.

If you have any other questions, please feel free to ask.

> Overall, thank you many times for doing this work! You even have Japanese UI elements that will make Thingweb project international ;)

I am glad to hear you say so.
I hope to take the next step to internationalize the client side as well.
If there are any odd English words, I would appreciate it if you could point them out to me.

**Comment by egekorkan** [comment]
@hidetak I have done the same thing once again and it worked, so I am sorry for the first comment about readproperty. I will now check the english and make commits if I see issues. If not, we can merge it and move to its new place

**Comment by egekorkan** [comment]
I made some small changes and it is looking good. Merging.

When moving to Eclipse, I will try to follow https://www.johno.com/move-directory-between-repos-with-git-history so that the contributors are credited correctly. I hope it works otherwise it would look as if I did all the work.

**Comment by hidetak** [comment]
@egekorkan 
> I made some small changes and it is looking good. Merging.

Thank you for the correction and merge.

**Comment by egekorkan** [comment]
In the end, I have used the instructions at https://gist.github.com/trongthanh/2779392

**Pull request merged**